Libraries/TranscriptTestRunner/TestRunner.cs

```diff
@@ -285,8 +291,17 @@ private async Task ExecuteTestScriptAsync(string callerName, CancellationToken c
             _logger.LogInformation($"\n------ Starting test {callerName} ----------");
 
             using var reader = new StreamReader(_testScriptPath);
+            var plainTestScript = await reader.ReadToEndAsync().ConfigureAwait(false);
+
+            if (_scriptParams?.Count > 0)
+            {
+                foreach (var param in _scriptParams)
+                {
+                    plainTestScript = plainTestScript.Replace("${" + param.Key + "}", param.Value);
```

Comment:
Can we look into using regex with multiple patterns so we traverse the script only once? (I'm not great with regex but "I think" this is doable, let me know if it isn't).

See if the following article helps: https://www.wipfli.com/insights/blogs/connect-microsoft-dynamics-365-blog/c-regex-multiple-replacements